### PR TITLE
SWATCH-2899: Subscription type is On-demand for Paygo products

### PR DESCRIPTION
**Pull request by wottop**
Jira issue: SWATCH-2899

## Description

The criteria for the On-Demand subscription type needs to be Prometheus enabled or Pay As You Go

### Setup

1. Start swatch-contracts
`SERVER_PORT=8003 QUARKUS_MANAGEMENT_PORT=9003 ./gradlew :swatch-contracts:quarkusDev`
2. Start the monolith
` DEV_MODE=true SERVER_PORT=8000 SPRING_PROFILES_ACTIVE=worker,kafka-queue,api,capacity-ingress,kafka-queue,rhsm-conduit ./gradlew :bootRun`
3. Create an entry in the org_config table for an org
`insert into org_config (org_id, opt_in_type,created, updated) values ('5345257','API','2024-06-21T12:50:39.496954Z','2024-06-21T12:50:39.496954Z')`
4. Run the opt-in
`http :8000/api/rhsm-subscriptions/v1/opt-in x-rh-identity:$(echo '{"identity":{"internal":{"org_id":"5345257"}}}' | base64 -w0)
`

### Steps

1. Run from command line to exercise endpoint [either v1 or v2 will work]
`http GET ':8003/api/rhsm-subscriptions/v2/subscriptions/products/ansible-aap-managed' x-rh-identity:$(echo '{"identity":{"type":"User","org_id":"5345257"}}' | base64 -w0)`
`http GET ':8000/api/rhsm-subscriptions/v1/subscriptions/products/ansible-aap-managed' x-rh-identity:$(echo '{"identity":{"internal":{"org_id":"5345257"}}}' | base64 -w0)`

### Verification

1. The subscription type will be shown as "On-demand" in all cases.

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/resource/api/v1/SubscriptionTableController.java [hunk 1]
```diff
@@ -195,7 +194,7 @@ public SkuCapacityReport capacityReportBySku( // NOSONAR
       }
     }
 
-    boolean isOnDemand = SubscriptionDefinition.isPrometheusEnabled(productId.toString());
+    boolean isOnDemand = productId.isPrometheusEnabled() || productId.isPayg();
```
Comment (Sgitario) [inline src/main/java/org/candlepin/subscriptions/resource/api/v1/SubscriptionTableController.java:197]: Moving this logic which is repeated in several places into the ProductId would avoid the duplication:
```java
public boolean isOnDemand() {
   return isPrometheusEnabled() || isPayg();
}
```
This way if in the future, the condition changes, we would only require to change in one place.

**Review by Sgitario — CHANGES_REQUESTED** [review]

**Review by diegomaranhao — APPROVED** [review]

**Comment by diegomaranhao** [comment]
Looks good.

**Review by Sgitario — APPROVED** [review]

**Pull request merged**